src/components/channel-page/ads.jsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+export default class Ads extends React.Component {
+
+  state = {
+    editIsPressed: false
+  }
+
+  toggleEditBtn = () => {
+    this.setState({ editIsPressed: !this.state.editIsPressed });
```

Comment:
this.setState(prevState => ({...prevState, editIsPressed: !prevState.editIsPressed}));

https://discuss.reactjs.org/t/setstate-function/5162